### DISCUSSION TITLE
efficiently print bytearray/bytes for communication

### DISCUSSION THREAD

**Discussion by Novakasa** [pybricks/support#621] [Q&A] [answered]
Hey all,
I've been wondering since the only way to send data to the PC is via `print`, how to efficiently send e.g. a bytearray to the PC. I assume that `print(data)` internally calls `str(data`), so the data is converted into a less efficient format:
![image](https://user-images.githubusercontent.com/59559612/151656142-d9f6967b-feed-4cc2-8630-91ae698a20f8.png)
Is there a way to prevent this, and just use up the 1033 bytes of bandwidth?

**Comment by Novakasa** [comment, marked as answer]
I think I might have found my answer:
```python
import usys
usys.stdout.buffer.write(bytearray(1000))
```

**Reply by dlech** [reply]
Bingo!
